src/core_plugins/input_control_vis/public/vis_controller.js

```diff
@@ -147,10 +147,11 @@ class VisController {
     this.vis.API.queryFilter.addFilters(newFilters, this.vis.params.pinFilters);
   }
 
-  clearControls = () => {
+  clearControls = async () => {
```

Comment:
It looks like InputControlVis is what calls this, and it doesn't look like it properly handles promise failures and the like (it seems to assume this is a synchronous method). I wonder if this may cause tricky to track down unhandled errors?

Reply:
What do you recommend? I could wrap it in a try/catch? `updateNestedControls` just calls `control.fetch` for each control that has ancestors. `control.fetch` should properly handle any exceptions and update the control state accordingly.